### ISSUE TITLE
Does not show data for 2021

### ISSUE THREAD

**Issue by niekniek89** [xirixiz/homeassistant-afvalwijzer#121]
## Version of the custom_component
5.2.0
-->

## Configuration
- platform: afvalwijzer # Required
  provider: mijnafvalwijzer
  postal_code: !secret postcode  # Required
  street_number: !secret housenumber # Required
Add your logs here.

```

## Describe the bug
Hello,

Could the integration not be showing dates for 2021 yet?
"Restafval" waste was collected last week, and the next date will not be until next year (2021).
```
![Schermafbeelding 2020-12-06 om 14 07 54](https://user-images.githubusercontent.com/48858098/101280963-9bb33080-37cc-11eb-9cfc-134ae2b983d2.png)

**Comment by xirixiz** [comment]
That's normal behaviour indeed. It's dependent of data made availble for your commune.

You can verify here if data for "2021" is available:
```
https://api.mijnafvalwijzer.nl/webservices/appsinput/?apikey=5ef443e778f41c4f75c69459eea6e6ae0c2d92de729aa0fc61653815fbd6a8ca&method=postcodecheck&postcode=<POSTCAL_CODE>&street=&huisnummer=<STREET_NUMBER>&toevoeging=&app_name=afvalwijzer&platform=phone&afvaldata=2020-12-06&langs=nl```

**Comment by Mariusthvdb** [comment]
sorry for commenting on a closed issue, but maybe some extra consideration on this:

wouldn't it be feasible to have the integration check for this? and, if no dates are set yet, have it show 'TBA' or something like that?

As it stands we can template that in Lovelace or the backend probably, but, just like the 'Geen' for non -pickup days, Tba for this period of the year would be very nice indeed.

<img width="419" alt="Schermafbeelding 2020-12-09 om 09 35 34" src="https://user-images.githubusercontent.com/33354141/101605294-4d877280-3a02-11eb-8c73-7bdf8555e698.png">

beaus with an unguarded template we now have this in the frontend... ;-)

**Comment by xirixiz** [comment]
If no data is available the value already ends up in "Geen". I think you can modify the template by using this value if no valid date has been found.

**Comment by Mariusthvdb** [comment]
just noticed indeed that:

<img width="978" alt="Schermafbeelding 2020-12-09 om 09 58 04" src="https://user-images.githubusercontent.com/33354141/101607627-16ff2700-3a05-11eb-826d-3b519ba53f67.png">

`is_collection_date_today: true` and state is 'Geen'

must admit I have never checked what the state is on a `is_collection_date_today: true` day, does it show the next-pickup date in that case?

If so, we should maybe check for '`is_collection_date_today: true` and 'Geen' combination to know now future dat has been set?

**Comment by xirixiz** [comment]
is_collection_date_today: true and state is 'Geen'

...this isn't correct indeed, let me go bugfixing 👍

**Comment by niekniek89** [comment]
Hello,

what am I doing wrong?
I used your piece of config ...

![image](https://user-images.githubusercontent.com/48858098/101617208-a4944400-3a10-11eb-8a9f-e3381278d018.png)

**Comment by Mariusthvdb** [comment]
I am no longer using the _next template sensors, but only the sensors the integration creates, not sure why you post the template sensor?

**Comment by niekniek89** [comment]
Hi, so I can remove all _next template sensors? I saw somewhere in a previous post that you used this ..

**Comment by Mariusthvdb** [comment]
if you dont need them anywhere else in your config or show them I Lovelace, yes you can. Simply comment them, so you always go back if so needed

**Comment by niekniek89** [comment]
I used this to create the above in lovelace .. but that's not necessary anymore? (the template sensors)

**Comment by Mariusthvdb** [comment]
nope, its the new integration causing issues...
today's a paper pickup

<img width="442" alt="Schermafbeelding 2020-12-09 om 11 54 18" src="https://user-images.githubusercontent.com/33354141/101620828-4b7adf00-3a15-11eb-9fe8-3653a5c5c1d1.png">

but I see 5.2.6 has arrived, Bram is in a whirlwind at the moment ;l-)
lets see if that fixes things

**Comment by niekniek89** [comment]
@Mariusthvdb thanks for your help. So I don't have to adjust anything to get it working (restafval) issue?

**Comment by Mariusthvdb** [comment]
this should fix it using 5.2.6:

```
  - type: custom:auto-entities
    card:
      type: entities
      title: Volgende ophaaldata
    filter:
      exclude:
        - entity_id: sensor.afvalwijzer*next*
        - entity_id: sensor.afvalwijzer*to*

      include:
        - entity_id: sensor.afvalwijzer_*
          options:
            type: custom:template-entity-row
            state: >
              {% set trash = config.entity %}
              {% set future = state_attr(trash,'days_until_collection_date') %}
              {% if future is none %} Tba
              {% else %}
              {{as_timestamp(strptime(states(trash),'%d-%m-%Y'))
                              |timestamp_custom('%-d %b')}}
              {% endif %}
            secondary: >
              {% set trash = config.entity %}
              {% set future = state_attr(trash,'days_until_collection_date') %}
              {% if future is none %} Nog niet gepubliceerd
              {% else %}
                {% set count = state_attr(trash,'days_until_collection_date')|int %}
                {% set day = as_timestamp(strptime(states(trash),'%d-%m-%Y'))
                   |timestamp_custom('%A') %}
                {% set dagen =
                  {'Monday': 'Maandag',
                  'Tuesday': 'Dinsdag',
                  'Wednesday': 'Woensdag',
                  'Thursday': 'Donderdag',
                  'Friday': 'Vrijdag',
                  'Saturday': 'Zaterdag',
                  'Sunday': 'Zondag'} %}
                {% set dag = dagen[day] if day in dagen else day %}
                {% set unit = 'Dag' if count == 1 else 'dagen' %}

                {% if count >= 14 %} {% set phrase = dag + ' over 2 weken' %}
                {% elif count >= 7 %} {% set phrase = 'Volgende week ' + dag %}
                {% elif count >= 3 %} {% set phrase = 'komende ' + dag %}
                {% elif count == 2 %} {% set phrase = dag + ', overmorgen' %}
                {% elif count == 1 %} {% set phrase = 'morgen, ' + dag %}
                {% else %} {% set phrase = 'Vandaag, ' + dag %}
                {% endif %}
                {{phrase}} {% if count != 0 %} ({{count}} {{unit}}) {% endif %}
              {% endif %}
    sort:
      method: state
      numeric: true
```

<img width="475" alt="Schermafbeelding 2020-12-09 om 12 11 44" src="https://user-images.githubusercontent.com/33354141/101622740-bb8a6480-3a17-11eb-9b81-e440dfbe639b.png">
